### DISCUSSION TITLE
Can't use components

### DISCUSSION THREAD

**Discussion by sudo-sand** [huntabyte/shadcn-svelte#107] [Help]
I really like this components, but for the general components like buttons this works. With something like Alert, I don't have access to any events or the ability to close it.

I have this code
```svelte
<script>
// ...
  const dispatch = createEventDispatcher()
  const toggle_is_open = () => is_open = false
  const handle_ok = () => {
    dispatch('ok')
    is_open = false
  }
</script>

<AlertDialog bind:open={is_open}>
```

I've been trying to get if the user actually pressed yes with no luck. The current way is it just closes the dialog without firing any events.
So I had to use a custom button to get `on:click` to work but I have this bug where I can't seem to close the `AlertDialog`.

**Comment by huntabyte** [comment]
If you could provide a stackblitz link to this I'd be more than happy to take a look!

**Reply by sudo-sand** [reply]
I tried to create repls using stackblitz, sveltelab.dev, and codesandbox. But they all failed 😅

I just want to know how to get an event from the AlertDialog? user clicked OK or Cancel.

**Comment by martinjo** [comment]
I added the **m-click** and **m-keydown** events to <AlertDialogPrimitive.Action> in **alert-dialog-action.svelte** and could then use those to repsond to the user action. Not sure if this is the way it's intended to be used. I was kind of expecting on:click to work 🤷‍♂️

<AlertDialogPrimitive.Action
	class={cn(buttonVariants(), className)}
	{...$$restProps}
	**on:m-click** 
	**on:m-keydown**
